lib/ansible/module_utils/ec2.py

```diff
@@ -41,9 +41,13 @@
 
 
 def boto3_conn(module, conn_type=None, resource=None, region=None, endpoint=None, **params):
-    profile = params.pop('profile_name', None)
-    params['aws_session_token'] = params.pop('security_token', None)
-    params['verify'] = params.pop('validate_certs', None)
+    # if params are inherited from get_aws_connection_info(), they are already properly set
+    if params['profile_name'] is None:
+        profile = params.pop('profile_name', None)
```

Comment:
I'm with @dblenkus it doesn't really make sense at all. If `params['profile_name']` is already `None` then why would you do this over the original line?